scripts/extract-codeblock.awk

```diff
@@ -0,0 +1,23 @@
+BEGIN {CODEBLOCK = 0; SNIPPET = 0; RED = "\\033[0;31m"; NO_COLOUR = "\\033[0m"; }
+BEGINFILE {
+    if (ERRNO != "") {
+        print "couldn't open " FILENAME " skipping"
+        nextfile
+    }
+}
+/^\[source,yaml\]/                           { SNIPPET = FNR; next} # code block follows between `----`
```

Comment:
I'd like to replace `yaml` here with `BLOCKTYPE` but I don't actually know how to do the pattern match against a variable.

Reply:
Does https://www.gnu.org/software/gawk/manual/gawk.html#Computed-Regexps help?